doc: Documentation for IPv6 and TLS migration (#1291)
apple/foundationdb: documentation/sphinx/source/tls.rst

Since version 6.1, FoundationDB clusters can be converted to TLS without downtime. FoundationDB server can listen to TLS and unencrypted traffic simultaneously on two separate ports. As a result, FDB clusters can live migrate to TLS:

1) Restart each FoundationDB server individually, but with an additional listen address for TLS traffic::

     /path/to/fdbserver -C fdb.cluster -p 127.0.0.1:4500 -p 127.0.0.1:4600:tls

   Since, the server still listens to unencrypted traffic and the cluster file still contains the old address, rest of the processes will be able to talk to this new process.

2) Once all processes are listening to both TLS and unencrypted traffic, switch one or more coordinator to use TLS. Therefore, if the old coordinator list was ``127.0.0.1:4500,127.0.0.1:4501,127.0.0.1:4502``, the new one would be something like ``127.0.0.1:4600:tls,127.0.0.1:4501,127.0.0.1:4502``. Switching few coordinators to TLS at a time allows a smoother migration and a window to find out clients who do not yet have TLS configured. The clients which cannot connect to TLS coordinators are listed in ``fdbstatus`` (look for ``tls_configured`` field in ``clients``)::

> 

Reviewer: Maybe add another paragraph before 3): 
If there exist a client that cannot connect to all coordinators after a coordinator is switched to TLS, it mean the client does not set up its TLS correctly. System operator should notify the client to correct the client's TLS configuration. 
Otherwise, when all coordinators are switched to TLS ports, the client will loose connection.

Reviewer: I will just suggest the change then.

Author: Thanks! I added that part to the new changes.

> 

Reviewer: This part is not consistent with the implementation.
It should be the `connected_coordinators ` field. 
Also it needs to mention that a client is TLS configured if the `connected_coordinators` value is the total number of coordinators.

> 

Author: Yes, my bad. You are right. I was just looking at old PR comments and forgot to update the field.

    "clients" : {
        "count" : 2,
        "supported_versions" : [
            {
                "client_version" : "6.1.0",
                "connected_clients" : [
                    {
                        "address" : "127.0.0.1:42916",

>                         "log_group" : "default",

Reviewer: ```suggestion
                        "connected_coordinators" : 2,
                        "log_group" : "default",
```

>                         "tls_configured" : false

Reviewer: ```suggestion

```

                    },
                    {
                        "address" : "127.0.0.1:42918",

>                         "log_group" : "default",

Reviewer: ```suggestion
                        "connected_coordinators" : 1,
                        "log_group" : "default",
```

>                         "tls_configured" : true

Reviewer: ```suggestion

```

>                         "tls_configured" : true

Reviewer: ```suggestion
```

                    }
                ]
            }, ...
        ]
    }

> 

Reviewer: ```suggestion
3) If there exist a client (e.g., the client 127.0.0.1:42918 in the above example) that cannot connect to all coordinators after a coordinator is switched to TLS, it mean the client does not set up its TLS correctly. System operator should notify the client to correct the client's TLS configuration.
Otherwise, when all coordinators are switched to TLS ports, the client will loose connection.
```

> 3) Repeat (2) until all the address in coordinator list are TLS.

Reviewer 2: I think this should be "addresses" instead of "address"

> 3) Repeat (2) until all the address in coordinator list are TLS.

Reviewer: ```suggestion
4) Repeat (2) until all the address in coordinator list are TLS.
Note: Make sure all clients can connect to all coordinators whenever you change a coordinator. 
```

> 4) Restart each FoundationDB server, but only with one public address that listens to TLS traffic only.

Reviewer: ```suggestion
5) Restart each FoundationDB server, but only with one public address that listens to TLS traffic only.
```
